### DISCUSSION TITLE
Please provide `ssh_known_hosts` for GH services in Actions

### DISCUSSION THREAD

**Discussion by mpdude** [community/community#26611] [Actions]
<p>Dear GitHub,</p>
<p>would it be possible to set up Action worker nodes in a way that we already have the SSH Host Keys for GitHub services (repos, Gists) etc. available in either <code>/etc/ssh/ssh_known_hosts</code> or <code>~/.ssh/known_hosts</code>? </p>
<p>That would make workflows somewhat more straightforward when we e. g. need to fetch dependencies.</p>
<p>Also, it would be more secure than having to run <code>ssh-keyscan</code> and just accept whatever keys we find. You should be in the best position to always provide the correct and up-to-date keys :wink:.</p>
<p>Thanks!</p>

**Comment by ethomson** [comment, marked as answer]
<p>Neat idea.  I can see how this could help.  We’ll take a look.</p>

**Comment by tonyarnold** [comment]
<p>Did this end up happening? I’m still finding that I have to run the following command from inside my workflows to use Swift Package Manager:</p>
<pre><code>for ip in $(dig @8.8.8.8 github.com +short); do ssh-keyscan github.com,$ip; ssh-keyscan $ip; done 2&gt;/dev/null &gt;&gt; ~/.ssh/known_hosts
</code></pre>

**Comment by 3a4oT** [comment]
<p>Hey <a class="mention" href="/tonyarnold">@tonyarnold</a>. Are you able to run Swift package tests with private dependencies? If so can you please share your approach since I can’t make it work for some reason GitHub runner do not use the package’s local <strong>.ssh</strong> folder and always fails to fetch dependencies. There is a <a href="https://github.community/t/github-action-with-private-swift-package/17362/6">related topic</a>.  Thanks!</p>

**Comment by tonyarnold** [comment]
<p>I am able to check out private Swift packages:</p>
<pre><code class="lang-yaml">- name: "Set up SSH agent"
  uses: webfactory/ssh-agent@v0.1.1
  with:
    ssh-private-key: ${{ secrets.CI_SSH_PRIVATE_KEY }}

- name: "Add GitHub to the SSH known hosts file"
  run: |
    for ip in $(dig @8.8.8.8 github.com +short); do \
      ssh-keyscan github.com,$ip; \
      ssh-keyscan $ip; \
    done 2&gt;/dev/null &gt;&gt; ~/.ssh/known_hosts
</code></pre>
<p>There is one other change you need to make if you’re using Xcode directly, rather than Swift Package Manager, and that is to pass the <code>-usePackageSupportBuiltinSCM</code> flag to <code>xcodebuild</code>.</p>
<p>Once you’ve done this, you should be fine!</p>

**Comment by saudeon** [comment]
<p><a class="mention" href="/tonyarnold">@tonyarnold</a>  I just want to thank you immensely.</p>
<p>Finding your post, and your comment in regards to the <code>-usePackageSupportBuiltinSCM</code> flag for <code>xcodebuild</code> finally resolved a huge headache with GitHub Actions.</p>

**Comment by BrentMifsud** [comment]
<p>I managed to fix this issue by using ssh for all my private packages and also adding the following to my GitHub action workflow file:</p>
<ol>
<li>
<code>-usePackageSupportBuiltinSCM</code> flag to Xcode build step</li>
<li>adding a run step to add the above SSH key prior to running Xcode build.</li>
</ol>
<pre><code class="lang-auto">- name: Add CI SSH Key
  run: ssh-add - &lt;&lt;&lt; "${{ secrets.CI_SSH_KEY }}"
</code></pre>

**Reply by Sandeeepp** [reply]
Hey @BrentMifsud This solution was working when I had only 1 swift package in my project. 
Any idea how to make this work with 2 swift packages.

I tried different solutions but actions fails saying
 `ERROR: Repository not found`
 `fatal: Could not read from remote repository.
    Please make sure you have the correct access rights
    and the repository exists.`

**Comment by dkulundzic** [comment]
<p><a class="mention" href="/tonyarnold">@tonyarnold</a> I’m using <code>fastlane</code>, do you know how to pass <code>-usePackageSupportBuiltinSCM</code> to <code>build_app</code>?</p>
<p>Currently, I have tried and failed :smiley:</p>
